Sources/Socket.swift

```diff
@@ -3226,38 +3268,44 @@ public class Socket: SocketReader, SocketWriter {
 		self.readBuffer.initialize(to: 0x0)
 		memset(self.readBuffer, 0x0, self.readBufferSize)
 		var address: Address? = nil
+
+		var recvFlags: Int32 = 0
+		if (self.readStorage.length > 0) {
+			recvFlags |= Int32(MSG_DONTWAIT)
+		}
 		
 		let addr = sockaddr_storage()
 		var length = socklen_t(MemoryLayout<sockaddr_storage>.size)
 		var addrPtr = addr.asAddr()
-		
+
 		// Read all the available data...
 		#if os(Linux)
-			let count = Glibc.recvfrom(self.socketfd, self.readBuffer, self.readBufferSize, 0, &addrPtr, &length)
+			let count = Glibc.recvfrom(self.socketfd, self.readBuffer, self.readBufferSize, recvFlags, &addrPtr, &length)
 		#else
-			let count = Darwin.recvfrom(self.socketfd, self.readBuffer, self.readBufferSize, 0, &addrPtr, &length)
+			let count = Darwin.recvfrom(self.socketfd, self.readBuffer, self.readBufferSize, recvFlags, &addrPtr, &length)
 		#endif
-		
+
 		// Check for error...
 		if count < 0 {
-			
+
 			// - Could be an error, but if errno is EAGAIN or EWOULDBLOCK (if a non-blocking socket),
 			//		it means there was NO data to read...
 			if errno == EAGAIN || errno == EWOULDBLOCK {
-				
-				return (0, address)
+
+				// FIXME: If we reach this point because data is available in the internal buffer, we will be *unable* to associate it with an address...
+				return (self.readStorage.length, nil)
```

Comment:
While it would be nice to have the feature, it's probably best to follow the normal datagram behavior and only return `bufSize` data regardless of the amount of data that's read in.  So, after you fix the current conflict for this PR and it gets merged, I'm going to remove the feature from the function.  So, this:
```
public func readDatagram(into buffer: UnsafeMutablePointer<CChar>, bufSize: Int, truncate: Bool = true )throws -> (bytesRead: Int, address: Address?)
```
will become this again...
```
public func readDatagram(into buffer: UnsafeMutablePointer<CChar>, bufSize: Int) throws -> (bytesRead: Int, address: Address?)
```
The `truncate` feature on the **TCP** version of this API will be left intact.  Both versions of this API are potentially useful given the proper scenario.

Finally, please hold off on any other PR's until I push the code for the above change *tomorrow* and I get a release tagged after it passes CI.  I'm about to start my *year end vacation* on Friday and want to wrap things up before then. I'll still be monitoring things during my vacation but most likely not actively working on it.  Thanks for your excellent work.